internal/search/query/predicate.go

```
@@ -500,3 +504,27 @@ func (f *FileHasOwnerPredicate) Unmarshal(params string, negated bool) error {
 
 func (f FileHasOwnerPredicate) Field() string { return FieldFile }
 func (f FileHasOwnerPredicate) Name() string  { return "has.owner" }
+
+type SymbolReferencesPredicate struct {
+	SymbolSearch string
+}
+
+func (s *SymbolReferencesPredicate) Unmarshal(params string, _ bool) error {
+	s.SymbolSearch = params
+	return nil
+}
+
+func (s SymbolReferencesPredicate) Field() string { return FieldSymbol }
+func (s SymbolReferencesPredicate) Name() string  { return "references" }
+
+type SymbolImplementsPredicate struct {
+	SymbolSearch string
+}
+
+func (s *SymbolImplementsPredicate) Unmarshal(params string, _ bool) error {
+	s.SymbolSearch = params
+	return nil
+}
```

Would it make sense to do the `query.Parse` here? That way we can return an error on query parsing rather than during execution.

---

I turned the `SymbolSearch` string fields into a `Plan`: 8a5bf2e70637f3fe0195110a6d67ba3978282280